grails-app/controllers/org/transmartproject/rest/ExportController.groovy

```diff
@@ -29,29 +29,27 @@ class ExportController {
      *  Returns datatypes and patient number of given concepts.
      *
      */
-    def datatypes(){
-        def jsonSlurper = new JsonSlurper()
+    def datatypes() throws NoSuchResourceException {
         if (!(params.containsKey('concepts'))){
-            throw new NoSuchElementException(
-                    "No parameter named concepts."
-            )
+            throw new NoSuchResourceException("No parameter named concepts was given.")
+        }
+        if (params.get('concepts') == "") {
+            throw new InvalidArgumentsException("Parameter concepts has no value.")
         }
-        def test = params.get('concepts').decodeURL()
+        def jsonSlurper = new JsonSlurper()
+        def conceptParameters = params.get('concepts').decodeURL()
         try {
-            def concept_arguments = jsonSlurper.parseText(test)
-            if (concept_arguments==null){
-                throw new NullArgumentException(
-                        "Parameter concepts has no value."
-                )
-            }
-            List datatypes = []
-            concept_arguments.each { it ->
+            def conceptArguments = jsonSlurper.parseText(conceptParameters)
+            List dataTypes = []
+            int cohortNumber = 1
+            conceptArguments.each { it ->
```

Comment:
There is quite some logics in this call, preferably you would put as much transformation and business logic in a Service. Leaving Http handling and top level control flow to the Controler